File: samples/three-layer-kmp-domain-simple/androidApp/src/main/kotlin/com/example/kmpsamples/threelayerkmpdomainsimple/android/TaskRepository.kt
package com.example.kmpsamples.threelayerkmpdomainsimple.android

import com.example.kmpsamples.threelayerkmpdomainsimple.shareddomain.Task
import com.example.kmpsamples.threelayerkmpdomainsimple.shareddomain.TaskRepository as SharedTaskRepository

class TaskRepository(
    private val dataSource: TaskDataSource
) : SharedTaskRepository {
    override fun loadTask(platform: String): Task = Task(
        title = "Android repository task",
        details = dataSource.loadTask(platform)
    )
}

File: samples/three-layer-kmp-domain-simple/iosApp/Sources/TaskRepository.swift
import SharedDomain

final class IosTaskRepository: TaskRepository {
    private let dataSource: TaskDataSource

    init(dataSource: TaskDataSource) {
        self.dataSource = dataSource
    }

    func loadTask(platform: String) -> SharedDomain.Task {
        SharedDomain.Task(
            title: "iOS repository task",
            details: dataSource.loadTask(platform: platform)
        )
    }
}
